app/queries/interests_query.rb

```diff
@@ -0,0 +1,21 @@
+class InterestsQuery
+  def self.call
+    new.call
+  end
+
+  def call
+    search_interest
+  end
+
+  private
+
+  def search_women_age_20_30
+    User.where(is_male: false)
+        .where(birthday: Date.current - 30.year..Date.current - 20.year)
```

Comment:
I would sugest adding parentheses `(...)` closure around `Date.current - 30.year` and `Date.current - 20.year` as I find seeing `30.year..Date.current` in the middle a little confusing.